Runtime/Model/MachineIdStorage.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Backtrace.Unity.Extensions;
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+using UnityEngine;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Model
+{
+    /// <summary>
+    /// Backtrace Machine Id storage
+    /// </summary>
+    internal class MachineIdStorage
+    {
+        /// <summary>
+        /// Player prefs machine identifier key
+        /// </summary>
+        internal const string MachineIdentifierKey = "backtrace-machine-id";
+
+        /// <summary>
+        /// Generate unique machine id. 
+        /// </summary>
+        /// <returns>Unique machine id Guid in a string format</returns>
+        internal string GenerateMachineId()
+        {
+            var storageMachineId = FetchMachineIdFromStorage();
+            if (!string.IsNullOrEmpty(storageMachineId))
+            {
+                return storageMachineId;
+            }
+#if !UNITY_WEBGL && !UNITY_SWITCH
+            var unityIdentifier = UseUnityIdentifier();
+            if (!GuidHelper.IsEmptyGuid(unityIdentifier))
+            {
+                StoreMachineId(unityIdentifier);
```

Comment:
Programming flow nitpicky-ness: `StoreMachineId` is called in multiple locations. Why not assign one `machineId` var and store it in the end instead of doing an early out? Downside is that you'll have a large if/elseif/elseif/ellse/ tree, so, you can probably ignore this.

Reply:
the logic would be harder to understand with more ifs to check. Now it's kinda clear.